src/monta_mcp/__init__.py

```
@@ -1,2 +1,398 @@
+    sdk_page = page - 1
+
+    # Parse and validate parameters
+    parsed_from_date: datetime | None = None
+    parsed_to_date: datetime | None = None
+    parsed_state: ChargeState | None = None
+
+    if from_date:
+        try:
+            parsed_from_date = datetime.fromisoformat(from_date.replace("Z", "+00:00"))
+        except ValueError as e:
+            raise ValueError(f"Invalid from_date format. Expected ISO format: {e}")
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)